distributed/worker_state_machine.py

```diff
@@ -353,8 +354,60 @@ class AddKeysMsg(SendMessageToScheduler):
 
 @dataclass
 class StateMachineEvent:
-    __slots__ = ("stimulus_id",)
+    __slots__ = ("stimulus_id", "handled")
     stimulus_id: str
+    #: timestamp of when the event was handled by the worker
+    # TODO switch to @dataclass(slots=True) and uncomment (requires Python >=3.10)
+    # handled: float | None = field(init=False, default=None)
+    _classes: ClassVar[dict[str, type[StateMachineEvent]]] = {}
+
+    def __new__(cls, *args, **kwargs):
+        self = object.__new__(cls)
+        self.handled = None
+        return self
+
+    def __init_subclass__(cls):
+        StateMachineEvent._classes[cls.__name__] = cls
+
+    def log(self, *, handled: float) -> StateMachineEvent:
```

Comment:
I think this could be named something more descriptive. How about one of the following?

- `logabble_event`?
- `to_loggable_event`?

Reply:
renamed